src/merkle.rs

```diff
@@ -286,12 +296,8 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
 
                 let layer: Vec<_> = data
                     .read_range(read_start..read_start + width)?
-                    .par_chunks(2)
-                    .map(|v| {
-                        let lhs = v[0].to_owned();
-                        let rhs = v[1].to_owned();
-                        A::default().node(lhs, rhs, level)
-                    })
+                    .par_chunks(branches)
+                    .map(|v| A::default().multi_node(v.to_vec(), level))
```

Comment:
might be worth special casing binary, given the perf hit this creates (not 100% sure though)